charts/clusterpedia/templates/apiserver-deployment.yaml

```diff
@@ -30,6 +30,32 @@ spec:
     spec:
       {{- include "clusterpedia.apiserver.imagePullSecrets" . | nindent 6 }}
       initContainers:
+    {{- if .Values.externalStorage.createDatabase }}
+      - name: ensure-database
+        image: {{ include "clusterpedia.storage.image" . }}
+        command:
+          - /bin/sh
+          - -ec
+          {{- if eq (include "clusterpedia.storage.type" .) "postgres" }}
+          - |
+            until psql -U {{ include "clusterpedia.storage.user" . }} -h {{ include "clusterpedia.storage.host" . }} -p {{ include "clusterpedia.storage.port" . }} postgres -c "SELECT 1 FROM pg_database WHERE datname = '{{ include "clusterpedia.storage.database" . }}'" | grep -q 1 || psql -U {{ include "clusterpedia.storage.user" . }} -h {{ include "clusterpedia.storage.host" . }} -p {{ include "clusterpedia.storage.port" . }} postgres -c "CREATE DATABASE {{ include "clusterpedia.storage.database" . }} owner {{ include "clusterpedia.storage.user" . }} " -c "GRANT ALL PRIVILEGES ON DATABASE {{ include "clusterpedia.storage.database" . }} to {{ include "clusterpedia.storage.user" . }}"; do 
+            echo waiting for database check && sleep 1;
+            done;
+            echo 'DataBase OK ✓'
+          {{- else if eq (include "clusterpedia.storage.type" .) "mysql" }}
+          - |
+            until mysql -u{{ include "clusterpedia.storage.user" . }} -p"${DB_PASSWORD}" --host={{ include "clusterpedia.storage.host" . }} --port={{ include "clusterpedia.storage.port" . }} -e 'CREATE DATABASE IF NOT EXISTS {{ include "clusterpedia.storage.database" . }}'; do
+            echo waiting for database check && sleep 1; 
+            done;
+            echo 'DataBase OK ✓'
+          {{- end }}
+        env:
+          - name: {{ include "clusterpedia.storage.password.name" . }}
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "clusterpedia.internalstorage.fullname" . }}
+                key: password
+    {{- end }}
       - name: check-storage
```

Comment:
`ensure-database` seems to have some overlap with `check-storage`. 
Could we not run the init container `check-storage` when the user needs to create a database?

Reply:
Or the init container `check-storage` is run first and only checks if mysql/postgres is ready,

The pg_isready connects to the postgres database, not to the user-specified database
>           - until pg_isready -U {{ include "clusterpedia.storage.user" . }} -d "dbname={{ include "clusterpedia.storage.database" . }}" -h {{ include "clusterpedia.storage.host" . }} -p {{ include "clusterpedia.storage.port" . }}; do sleep 1; done

Reply:
Maybe we can run only 1 container, if `ensure-database` is rendered, `check-storage` can be replaced.